qiskit/utils/mitigation/fitters.py

```diff
@@ -0,0 +1,439 @@
+                raise QiskitError("Qubit not in the original set of qubits")
+
+        # build state labels
+        new_state_labels = count_keys(len(qubit_sublist))
+
+        # mapping between indices in the state_labels and the qubits in
+        # the sublist
+        qubit_sublist_ind = []
+        for sqb in qubit_sublist:
+            for qbind, qubit in enumerate(self._qubit_list):
+                if qubit == sqb:
+                    qubit_sublist_ind.append(qbind)
+
+        # states in the full calibration which correspond
+        # to the reduced labels
+        q_q_mapping = []
+        state_labels_reduced = []
+        for label in self.state_labels:
+            tmplabel = [label[index] for index in qubit_sublist_ind]
+            state_labels_reduced.append("".join(tmplabel))
+
+        for sub_lab_ind, _ in enumerate(new_state_labels):
+            q_q_mapping.append([])
+            for labelind, label in enumerate(state_labels_reduced):
+                if label == new_state_labels[sub_lab_ind]:
+                    q_q_mapping[-1].append(labelind)
+
+        new_fitter = CompleteMeasFitter(
+            results=None, state_labels=new_state_labels, qubit_list=qubit_sublist
+        )
+
+        new_cal_matrix = np.zeros([len(new_state_labels), len(new_state_labels)])
+
+        # do a partial trace
+        for i in range(len(new_state_labels)):
+            for j in range(len(new_state_labels)):
+
+                for q_q_i_map in q_q_mapping[i]:
+                    for q_q_j_map in q_q_mapping[j]:
+                        new_cal_matrix[i, j] += self.cal_matrix[q_q_i_map, q_q_j_map]
+
+                new_cal_matrix[i, j] /= len(q_q_mapping[i])
+
+        new_fitter.cal_matrix = new_cal_matrix
+
+        return new_fitter
+
+    def readout_fidelity(self, label_list=None):
+        """
+        Based on the results, output the readout fidelity which is the
+        normalized trace of the calibration matrix
+
+        Args:
+            label_list (bool): If `None`, returns the average assignment fidelity
+                of a single state. Otherwise it returns the assignment fidelity
+                to be in any one of these states averaged over the second
+                index.
+
+        Returns:
+            numpy.array: readout fidelity (assignment fidelity)
+
+        Additional Information:
+            The on-diagonal elements of the calibration matrix are the
+            probabilities of measuring state 'x' given preparation of state
+            'x' and so the normalized trace is the average assignment fidelity
+        """
+        return self._tens_fitt.readout_fidelity(0, label_list)
+
+
+class TensoredMeasFitter:
+    """
+    Measurement correction fitter for a tensored calibration.
+    """
+
+    def __init__(
+        self,
+        results,
+        mit_pattern: List[List[int]],
+        substate_labels_list: List[List[str]] = None,
+        circlabel: str = "",
```

Comment:
yeah I removed type hints wherever there were circular dependency issues or similar problems. Since this isn't user facing and likely won't get updated or called anywhere else I didn't think it was necessary to be fully type hinted vs just having it.